src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -585,6 +586,7 @@ private void IdentifyUserFunctions(ParseNode node)
                 context.UserFunctions.GetUserFunction(funcIdentifier, storageType == StorageModifier.GLOBAL ? (Int16)0 : GetContainingScopeId(node), node);
             int expressionHash = ConcatenateNodes(bodyNode).GetHashCode();
             userFuncObject.GetUserFunctionOpcodes(expressionHash);
+            userFuncObject.IsFunction = !isLock;
```

Comment:
All I did was replicate the logic in `PreProcessUserFunctionStatement`, which uses the same inverted logic (I suspect as a legacy from some old system).  As far as I can tell that is the only other time that `IsFunction` is even set.

I'd be OK with some kind of a flip if that makes it read cleaner to you, but your example would still need a `!` in front of `isLock`.  We could instead store the variable returned by `IsDefineFunctionStatement` and set `IsFunction` based on that.